src/visitAllChildren/index.ts

```diff
@@ -16,8 +10,13 @@ const getVisitor = (visitors: VisitorsMap) => {
     const allNodesVisitor = visitors.all;
     allNodesVisitor && allNodesVisitor(node);
 
+    if (ts.isInterfaceDeclaration(node)) {
+      node;
+    }
+
     const syntaxKindVisitor = visitors[node.kind];
-    syntaxKindVisitor && syntaxKindVisitor(node);
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    syntaxKindVisitor && syntaxKindVisitor(node as any);
```

Comment:
Since specific nodes are more specific types than `ts.Node`, sometimes the compiler correctly complains that `ts.Node` cannot be used as a specific node, for example `ts.InterfaceDeclaration`, because some fields are missing in `ts.Node`.

By casting node as `any`, we bypass the compiler at this point but gain no more specific types at the visitor level.

Reply:
Is there no way to resolve the mismatch by casting `node` to the type implied by `node.kind`?
If not, then I agree that `any` is the only possibility.

Reply:
I agree that casting it as `any` isn't so clean.

I'll try to solve it using [type guards](https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards).

The type guards are already defined by the TypeScript compiler API, for example:
```typescript
ts.isInterfaceDeclaration(node) && node.members.forEach(...)
```

In this example, if the type guard `ts.isInterfaceDeclaration()` is true, `node` is casted to the type `InterfaceDeclaration`. We can see that it's possible to use interface specific properties such as `members`.

Reply:
I created another pull request for this in order to discuss it there:

https://github.com/proglang/ts-ast-utils/pull/3